distributed/comm/utils.py

```diff
@@ -62,6 +63,8 @@ def _from_frames():
 
     if deserialize and size > FRAME_OFFLOAD_THRESHOLD:
         res = yield offload(_from_frames)
+        # offload() may result in cyclic references
+        gc.collect()
```

Comment:
Shouldn't this be added to `_to_frames` as well? I think the same problem could occur in the other direction.

Reply:
True, we need it here as well. I will try to work on a solution that could work for all offset() calls.